### PR TITLE
React1

### PR DESCRIPTION
**Pull request by ijeomaodinko**
**Description**
This React capstone project ([remember what are they?](https://github.com/microverseinc/curriculum-html-css/blob/main/articles/capstone_intro.md))is about building a mobile web application to check a list of metrics (numeric values) that you will create making use of React and Redux.

You will select an API that provides numeric data about a topic that you like and then build the webapp around it. The webapp will have several pages:

one page with a list of items that could be filtered by some parameters; in the example below, it's a list of metrics that can be filtered by the country (imagine a search field to introduce the country name like Italy, Croatia, etc.). This page should be your homepage.
one page for the item details; in the example, the detail page for Czech Republic cities with number of views.
Home page

When the page loads, the webapp shows the list of categories that could be filtered out by some parameter, for example by the category name.
Along with the category name, you will display some numeric values per category that come from the API.
When the user clicks (or taps) on a category item, the application navigates to the details page.
Details page
In the details page, the webapp retrieves data from the API to show detailed data in the category.
When the user clicks on the "Back" button (<), the user navigates to the home page.

The link is the page deployment but is still in progress. Thanks

https://www.loom.com/share/5a3d5650f2fd4c5ea1a2850ca7c7af1d

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,70 +1,90 @@
...
+- CSS
+- Redux
+- VsCode
 
-The page will reload when you make changes.\
-You may also see any lint errors in the console.
+## Live Demo (if available)
 
-### `npm test`
+[Live Demo Link](https://livedemo.com)
```
Comment (Deepakdanger) [inline README.md]: - [ ] Kindly add the proper link to the live demo.
 - The student deploys the application using Heroku, Netlify or GitHub Pages.

File: src/components/Detail.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+import back from './assets/back.png';
+import './Detail.css';
+
+const Detail = () => {
+  const { id } = useParams();
+  const dispatch = useDispatch();
+  const coin = useSelector((state) => state.detailReducer);
+  useEffect(() => {
+    dispatch(assetSingle(id));
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Deepakdanger) [inline src/components/Detail.js]: - [ ] KIndly remove this line & solve the linter manually.

File: src/components/Home.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import { assetItem, assetFilter } from '../redux/Home';
+import Asset from './Asset';
+import './Home.css';
+
+const Home = () => {
+  const dispatch = useDispatch();
+  const assetList = useSelector((state) => state.assetReducer);
+  useEffect(() => {
+    dispatch(assetItem());
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (Deepakdanger) [inline src/components/Home.js]: - [ ] KIndly remove this line & solve the linter manually.

**Review by Deepakdanger — CHANGES_REQUESTED** [review]
Hi @ijeomaodinko, 

Good job so far! 👏
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight :100: 
- No linter error :heavy_check_mark: 
- Website looks professional :+1: 

## Required Changes ♻️
- [ ] Whenever i search anything, instead of mathing tiles ,I got this empty tile.KIndly fix this.
Screenshot from 2022-09-10 01-17-48

- [ ] _Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by ijeomaodinko** [comment]
The required changes are fixed.
The tile is showing information.
The commands were removed.
The site is showing

File: src/components/Home.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+  return (
+    <section className="sectionDiv">
+      <div className="optionDiv">
+        <div className="option">
+          <select className="desc" onChange={changeHandler}>
+            <option value="price"> Rank </option>
+            <option value="name"> Name </option>
+            <option value="vol"> High Price </option>
+          </select>
```
Comment (zhadier) [inline src/components/Home.js]: - [ ] Kindly ensure that you keep the search bar from the previous review, as that is the project requirement, to be able to filter using a specific category value, like the name. This functionality is quite easy to implement and can be found in most websites so learning how to create a search bar is imperative. You may use the following code as a guide to filter the list of bit coins you get from the reducer using the input value of a search bar. Kindly implement the search bar.

``` js
 let filteredList = bitcoinsList;
 const handleSearch = (evt) => {
    setSearch(evt.target.value); // State to control the input of the search bar
    if (evt.target.value.trim().length === 0) {
      filteredList = bitcoinsList;
    } else {
      const val = evt.target.value.trim().toLowerCase();
      filteredList = bitcoinsList.filter((bitcoin) => bitcoin.name.toLowerCase().includes(val));
    }
  };
```

File: src/redux/Detail.js [hunk 5]
```diff
@@ -0,0 +1,29 @@
+const apiUrl = 'https://api.coincap.io/v2/assets/';
+
+const FETCH_EACH = 'FETCH_EACH';
+// const initialState = [{ id: 'bitcoin', symbol: 'BTC' }];
```
Comment (zhadier) [inline src/redux/Detail.js]: - [ ] Kindly remove commented out code from your repo as it is considered bad for maintainability and manageability.

**Review by zhadier — CHANGES_REQUESTED** [review]
# Hi @ijeomaodinko  👋  

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet. 🌤️ 



 ## Highlights

✔️ No linter errors
✔️ GitFlow was setup correctly
✔️ Professional pull request title and description

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question @zhadier so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/App.js [hunk 6]
```diff
@@ -1,24 +1,21 @@
-import logo from './logo.svg';
-import './App.css';
+import React from 'react';
+import { BrowserRouter as Router, Routes, Route } from 'react-router-dom';
+// import { BrowserRouter, Routes, Route } from 'react-router-dom';
```
Comment (ahmetbozaci) [inline src/App.js]: - [ ] Like the last code reviewer said kindly remove commented code to make clean code.

**Review by ahmetbozaci — CHANGES_REQUESTED** [review]
# ♻️♻️♻️ STATUS: Changes Required  ♻️♻️♻️ 


@ijeomaodinko  ✋🏼 
#### Great work on making the changes requested by a previous reviewer 👏 
#### You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.
___ 

### 💯  Highlights 💯
- You set up linter files and no errors. ✔️
- Your pull request is very professional. It has a meaningful title and summary. Keep it up.✔️
___

### ♻️ Request changes ♻️

- [ ] _Check the comments under the review._

___

### Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
___

### Cheers and Happy coding! 💻 👏
almost

___

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification._

_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._
___

_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form.](https://airtable.com/shrQAqnBwek5a0O0s)_

[GitHub](https://github.com/ahmetbozaci) | [LinkedIn](https://linkedin.com/in/ahmetbozaci) | [Twitter](https://twitter.com/ahmtbozaci)

_Reviewed by [Ahmet Bozacı ](https://github.com/ahmetbozaci)_

**Review by emmanuellekamwa — APPROVED** [review]
**STATUS: APPROVED 🍾✔🎉**
Hello @ijeomaodinko 👋
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights

- You documented your work professionally.✔
- Your linters are passing all the checks
- You asked for the changes in a polite and friendly way.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**